server/cache/dirtools/dirtools.go

```diff
@@ -920,7 +984,7 @@ func (ff *BatchFileFetcher) FetchFiles(filesToFetch FileMap, opts *DownloadTreeO
 		if len(req.Digests) > 0 {
 			reqCopy := req
 			eg.Go(func() error {
-				return ff.batchDownloadFiles(ctx, reqCopy, filesToFetch, opts)
+				return ff.batchDownloadFiles(ctx, reqCopy, ff.filesToFetch, opts)
```

Comment:
could filesToFetch be removed from the batchDownloadFiles() method and refer to ff.filesToFetch internally?